dodiscover/constraint/skeleton.py

```diff
@@ -263,10 +280,12 @@ def fit(self, context: Context) -> None:
         context : Context
             A context object.
         """
+        self.context = context.copy()
```

Comment:
@darthtrevino I assume this doesn't need to be a [deep copy](https://www.geeksforgeeks.org/copy-python-deep-copy-shallow-copy/)?

Reply:
This creates a deep-copy of the necessary components via the `def copy(self)` function inside `Context` class.

Reply:
FWIW if you treat data structures as immutable, and use immutability-safe patterns like structural sharing for mutation, you can avoid deep-copy situations. Are there any good immutability libraries in the Python ecosystem?